manifest.go

```diff
@@ -156,6 +158,49 @@ func validateManifest(s string) ([]error, error) {
 	return warns, nil
 }
 
+// ValidateProjectRoots validates the project roots present in manifest.
+func ValidateProjectRoots(c *Ctx, m *Manifest, sm gps.SourceManager) error {
+	projectRoots := make([]gps.ProjectRoot, 0, len(m.Constraints)+len(m.Ovr))
+	for pr := range m.Constraints {
+		projectRoots = append(projectRoots, pr)
+	}
+	for pr := range m.Ovr {
+		projectRoots = append(projectRoots, pr)
+	}
+
+	// Channel to receive all the errors
+	errorCh := make(chan error, len(projectRoots))
+
+	var wg sync.WaitGroup
+	for _, pr := range projectRoots {
+		wg.Add(1)
+		go func(pr gps.ProjectRoot) {
+			defer wg.Done()
+			origPR, err := sm.DeduceProjectRoot(string(pr))
+			if err != nil {
+				errorCh <- err
+			} else if origPR != pr {
+				errorCh <- fmt.Errorf("the name for %q should be changed to %q", pr, origPR)
+			}
+		}(pr)
```

Comment:
If you define this function ahead of time, then you can launch the go routines from the two separate loops over the original slices, and avoid building the `projectRoots` slice and looping over them again.

Reply:
That's a great idea 😃 Thanks!